### ISSUE TITLE
[3DS] Sonic Advance 3 Extra Zone softlocked

### ISSUE THREAD

**Issue by TwystedXYZ** [mgba-emu/mgba#2977]
Release: Latest, stable.
System: New Nintendo 2DS XL
Game: Sonic Advance 3 (USA)
Desc.:
During the Final Zone, Nonaggression, Eggman refuses to charge the energy fully, remaining stuck at a specific frame of his attack. I tried both with and without cheat codes, still nothing. Endrift, I gave you the save file on Discord.

**Comment by endrift** [comment]
[Sonic_Advance_3.zip](https://github.com/mgba-emu/mgba/files/12065196/Sonic_Advance_3.zip)

Uploading the save game/states here so I don't lose track of them.

**Comment by ChanseyIsTheBest** [comment]
This probably isn't it but I would double check the ROM you are using. Sonic Advance 3 has an infamously buggy prototype that has some issues the final game doesn't have and this may be one of them.

**Comment by endrift** [comment]
I've confirmed how the Nonaggression boss is supposed to work on hardware, using this save file. I had to pull out my GameShark just so I had enough lives to not have to redo all of navigation to get to the Final Zone after dying over and over...I don't think I like this game.

Anyway, now that I've finally gotten around to seeing how it works on hardware I can try comparing to see what goes wrong in emulator.
